src/core/utils/modal.tsx

```diff
@@ -128,30 +116,36 @@ export type GuardedOpenModalProps = {
   };
 };
 
-// Redirect anonymous users to the login platform, including a return link
-// to this page with an open modal.
-export function guardedOpenModal({
-  authUrl,
-  modalId,
-  trackOnlineView,
-  open
-}: GuardedOpenModalProps) {
-  if (userIsAnonymous()) {
-    const returnUrl = currentLocationWithParametersUrl({
-      modal: modalId
-    });
-    redirectToLoginAndBack({
-      authUrl,
-      returnUrl,
-      trackingFunction: trackOnlineView
-    });
-    return;
-  }
-  // If user is not anonymous we just open the given modal + potentially track it.
-  if (trackOnlineView) {
-    trackOnlineView();
-  }
-  open(modalId);
-}
+export const useModalButtonHandler = () => {
+  const dispatch = useDispatch();
+  return {
+    open: (modalId: ModalId) => {
+      return dispatch(openModal({ modalId }));
+    },
+    close: (modalId: ModalId) => {
+      return dispatch(closeModal({ modalId }));
+    },
+    guardedOpenModal: (props: GuardedOpenModalProps) => {
```

Comment:
I don't think it is necesssary to suffix the function with "modal". It is already in the "modal scope". Otherwise we should rename the other functions to: `openModal` and `closeModal` which I find redundant.

I think you could destruct the props like we do in most cases. So it would be:
```typescript
openGuarded: ({modalId, authUrl, trackOnlineView}: GuardedOpenModalProps) => {